widget/ui/src/components/Toast/Toast.Provider.tsx

```diff
@@ -0,0 +1,84 @@
+import type {
+  ProviderContext,
+  ProviderPropTypes,
+  ToastProps,
+  ToastType,
+} from './Toast.types';
+import type { PropsWithChildren } from 'react';
+
+import React, { createContext, useContext, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+import { Toast } from './Toast';
+import { Wrapper } from './Toast.styles';
+
+const ToastContext = createContext<ProviderContext | undefined>(undefined);
+
+let id = 1;
+
+export const ToastProvider = (props: PropsWithChildren<ProviderPropTypes>) => {
+  const {
+    children,
+    anchorOrigin = {
+      horizontal: 'right',
+      vertical: 'bottom',
+    },
+    container,
+    style = {},
+  } = props;
+  const [toasts, setToasts] = useState<ToastProps[]>([]);
+
+  // eslint-disable-next-line react/jsx-no-constructed-context-values
+  const api: ProviderContext = {
+    addToast(content: ToastType) {
+      setToasts((toasts) => [
+        ...toasts,
+        {
+          ...content,
+          id: content.id || id++,
+          horizontal: anchorOrigin.horizontal,
+        },
+      ]);
+    },
+    removeToast(id: number | string) {
+      setToasts((toasts) => toasts.filter((t) => t.id !== id));
+    },
+  };
+
+  const uniqueIds = new Set();
+  const filteredToasts = toasts.filter((toast) => {
+    if (!uniqueIds.has(toast.id)) {
+      uniqueIds.add(toast.id);
+      return true;
+    }
+    return false;
+  });
+
+  return (
+    <ToastContext.Provider value={api}>
+      {children}
+      {container &&
+        createPortal(
+          <Wrapper
```

Comment:
It's better to rename Wrapper to ToastContainer and `style` to `containerStyle` to differentiate Toast and its container.

Reply:
fixed.